x-pack/functionbeat/provider/aws/aws/transformer/transformer.go

```diff
@@ -82,6 +86,59 @@ func KinesisEvent(request events.KinesisEvent) []beat.Event {
 	return events
 }
 
+func CloudwatchKinesisEvent(request events.KinesisEvent, compressed bool) ([]beat.Event, error) {
+	var evts []beat.Event
+	for _, record := range request.Records {
+		envelopeFields := common.MapStr{
+			"event_id":                record.EventID,
+			"event_name":              record.EventName,
+			"event_source":            record.EventSource,
+			"event_source_arn":        record.EventSourceArn,
+			"event_version":           record.EventVersion,
+			"aws_region":              record.AwsRegion,
+			"kinesis_partition_key":   record.Kinesis.PartitionKey,
+			"kinesis_schema_version":  record.Kinesis.KinesisSchemaVersion,
+			"kinesis_sequence_number": record.Kinesis.SequenceNumber,
+			"kinesis_encryption_type": record.Kinesis.EncryptionType,
+		}
+
+		kinesisData := record.Kinesis.Data
+		if compressed {
+			inBuf := bytes.NewBuffer(record.Kinesis.Data)
+			r, err := gzip.NewReader(inBuf)
+			if err != nil {
+				return nil, err
+			}
+
+			var outBuf bytes.Buffer
+			_, err = io.Copy(&outBuf, r)
+			if err != nil {
+				r.Close()
+				return nil, err
+			}
+
+			err = r.Close()
+			if err != nil {
+				return nil, err
+			}
+			kinesisData = outBuf.Bytes()
+		}
```

Comment:
Yes, that is what I expected also. (That's why the development too more than expected.) But when I tested it with a Kinesis stream receiving events from a Log stream, I got gzipped data. I tried to find documentation and all I got is this:
> You can use CloudWatch Logs subscription feature to stream data from CloudWatch Logs to Kinesis Data Firehose. All log events from CloudWatch Logs are already compressed in gzip format, so you should keep Firehose’s compression configuration as uncompressed to avoid double-compression. 
But this is about Data Firehose and I used Data Streams, so I guess it applies to all of Kinesis services.

I will spend a bit more time about finding out more about compression in Kinesis and Cloudwatch.

Reply:
As a compromise, I added a new option `base64_encoded` so we support both my experience with Kinesis and the documentation.